makefiles/boot/riotboot.mk

```diff
@@ -80,7 +80,8 @@ riotboot/bootloader/%: $(BUILDDEPS)
 	$(Q)/usr/bin/env -i \
 		QUIET=$(QUIET) PATH="$(PATH)"\
 		EXTERNAL_BOARD_DIRS="$(EXTERNAL_BOARD_DIRS)" BOARD=$(BOARD)\
-		DEBUG_ADAPTER_ID=$(DEBUG_ADAPTER_ID)\
+		DEBUG_ADAPTER_ID=$(DEBUG_ADAPTER_ID) \
+		PROGRAMMER=$(PROGRAMMER) \
```

Comment:
I think I came across this problem a long time ago. I think it's likely that more variables might need exporting. We could instead have something like:

```
RIOTBOOT_BOOTLOADER_VARS = \
    BOARD \
    EXTERNAL_BOARD_DIRS \
    DEBUG_ADAPTER_ID \
    PROGRAMMER \
    #
RIOTBOOT_BOOTLOADER_ENV_VARS := $(strip $(foreach varname,$(RIOTBOOT_BOOTLOADER_VARS ), \
    $(if $(filter file environment command,$(origin $(varname))), \
    $(varname)="$($(varname))",)))
```

But dismiss if you want, and trigger the ci.

Reply:
I agree that a more long term solution is needed. Maybe keep this for a follow-up ? I triggered the CI.